app/routes.php

```diff
@@ -377,6 +377,30 @@ function( Application $app, Request $request ) use ( $ffFactory ) {
 );
 
 $app->post(
+	'show-donation-confirmation',
```

Comment:
It has to accept GET as well as POST. But you're right, there are smarter ways to do this apart from copying all the stuff.